internal/encoding/merge/merge.go

```diff
@@ -1,30 +1,56 @@
+// Copyright 2019-2020 Grabtaxi Holdings PTE LTE (GRAB), All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be found in the LICENSE file
+
 package merge
 
 import (
 	"bytes"
+	"encoding/json"
 	"strings"
 	"sync"
-
-	"github.com/kelindar/talaria/internal/encoding/block"
-	"github.com/kelindar/talaria/internal/encoding/typeof"
-	"github.com/kelindar/talaria/internal/monitor/errors"
 )
 
 // Func represents merge function
-type Func func([]block.Block, typeof.Schema) ([]byte, error)
+type Func func(interface{}) ([]byte, error)
```

Comment:
Why do we need to change the merge function?

Reply:
Using it for bigquery sink.

Reply:
https://github.com/kelindar/talaria/pull/86/commits/4296dccb169ef54919accb14a1283d04135414d5#diff-9bbcd19d85e2be1b1cd890773d61e59ac7f33b9ad8908cdfe6c0bb7278f9966dR68

Reply:
@atris @kelindar  we have changed somethings in merge package, the orc and parquet. Just a gist of it, merge package will merge blocks with same schema if schema differers within []blocks sent for compaction it fails.
Moreover, i really don't understand why we have test cases around merge with different schema. Is this intended?